src/optimization-engine/Deploy-AzureOptimizationEngine.ps1

```diff
@@ -0,0 +1,1532 @@
+    }
+    if ($deploymentOptions["WorkspaceReuse"] -eq "y")
+    {
+        if (-not($deploymentOptions["WorkspaceName"]))
+        {
+            throw "WorkspaceName is required for silent deployment when WorkspaceReuse is set to 'y'."
+        }
+        if (-not($deploymentOptions["WorkspaceResourceGroupName"]))
+        {
+            throw "WorkspaceResourceGroupName is required for silent deployment when WorkspaceReuse is set to 'y'."
+        }
+    }
+    if (-not($deploymentOptions["DeployWorkbooks"]) -or ($deploymentOptions["DeployWorkbooks"] -ne "y" -and $deploymentOptions["DeployWorkbooks"] -ne "n"))
+    {
+        throw "DeployWorkbooks set to 'y' or 'n' is required for silent deployment."
+    }
+    if (-not($deploymentOptions["SqlAdmin"]))
+    {
+        throw "SqlAdmin is required for silent deployment."
+    }
+    if (-not($deploymentOptions["SqlPass"]))
+    {
+        throw "SqlPass is required for silent deployment."
+    }
+    if (-not($deploymentOptions["TargetLocation"]))
+    {
+        throw "TargetLocation is required for silent deployment."
+    }
+    if (-not($deploymentOptions["DeployBenefitsUsageDependencies"]))
+    {
+        throw "DeployBenefitsUsageDependencies is required for silent deployment."
+    }
+    if ($deploymentOptions["DeployBenefitsUsageDependencies"] -eq "y")
+    {
+        if (-not($deploymentOptions["CustomerType"]))
+        {
+            throw "CustomerType is required for silent deployment when DeployBenefitsUsageDependencies is set to 'y'."
+        }
+        if (-not($deploymentOptions["BillingAccountId"]))
+        {
+            throw "BillingAccountId is required for silent deployment when DeployBenefitsUsageDependencies is set to 'y'."
+        }
+        if (-not($deploymentOptions["CurrencyCode"]))
+        {
+            throw "CurrencyCode is required for silent deployment when DeployBenefitsUsageDependencies is set to 'y'."
+        }
+        if ($deploymentOptions["CustomerType"] -eq "MCA")
+        {
+            if (-not($deploymentOptions["BillingProfileId"]))
+            {
+                throw "BillingProfileId is required for silent deployment when CustomerType is set to 'MCA'."
+            }
+        }
+    }
+}
+
+if ((Test-Path -Path $lastDeploymentStatePath) -and !$silentDeploy)
+{
+    $depOptions = Get-Content -Path $lastDeploymentStatePath | ConvertFrom-Json
+    Write-Host $depOptions -ForegroundColor Green
+    $depOptionsReuse = Read-Host "Found last deployment options above. Do you want to repeat/upgrade last deployment (Y/N)?"
+    if ("Y", "y" -contains $depOptionsReuse)
+    {
+        foreach ($property in $depOptions.PSObject.Properties)
+        {
+            $deploymentOptions[$property.Name] = $property.Value
+        }    
+    }
+}
+
+$GitHubOriginalUri = "https://raw.githubusercontent.com/microsoft/finops-toolkit/main/src/optimization-engine/azuredeploy.bicep"
```

Comment:
We can account for this as part of moving this into the toolkit PowerShell modeule, but we should work with releases rather than direct code references so we can track downloads.

Reply:
I need guidance on how to convert these code references into releases. Anyways, the URL above can be overridden by  a script parameter.